src/components/editor/diacritics/DiacriticsSelection.jsx

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Stanford University see LICENSE for license
+
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { closeDiacritics } from 'actions/index'
+import shortid from 'shortid'
+import CharacterButton from './CharacterButton'
+import VocabChoice from './VocabChoice'
+import specialcharacters from '../../../../static/specialcharacters.json'
+
+
+const DiacriticsSelection = () => {
+  const dispatch = useDispatch()
+  const [characterButtons, setCharacterButtons] = useState([])
+  const show = useSelector(state => state.selectorReducer.editor.diacritics.show)
+  const languageHandler = useSelector(state => state.selectorReducer.editor.diacritics.languageHandler)
+
+  let cssClasses
+  if (show) {
+    cssClasses = ['sticky-top', 'bg-white']
+  } else {
+    cssClasses = ['d-none']
+  }
+
+  const selectVocabulary = (event) => {
+    const vocabChoice = event.target.value
+    if (vocabChoice === undefined) return null
+    const buttons = []
+    specialcharacters[vocabChoice].characters.forEach((char) => {
+      buttons.push(<CharacterButton key={shortid.generate()}
```

Comment:
Why not just use the character as the key?

Reply:
In some of the vocabularies, the character is not unique.